#**코드업 기초 100제**  
**C언어로 제작된 기초 100문제이나, 파이썬 개발자들을 위해 제작하였음.**  

블로그 : [세상의 잣대가 아닌](https://blog.naver.com/dnfla420)  
유튜브 : [우리밋](https://www.youtube.com/channel/UCS0F25vig_sPIQXMiK8IdSg?view_as=subscriber)

by.*우리밋*

---
## **1. 기초-출력**
---


**[1]**  
Hello  
출력하기

In [ ]:
print('Hello')

Hello


**[2]**  
Hello World  
출력하기

In [ ]:
print('Hello World')

Hello World


**[3]**  
Hello  
World  
(두 줄에 걸쳐 줄을 바꿔 출력하기)

**Tip::**  
1. 따옴표를 '''(작은따옴표) 혹은 """(큰따옴표) 와 같이 3개씩 사용하면 입력하는 모양대로 출력된다.
2. 처음 따음표 ''' 뒤에 역슬래기(\)를 해주어야 직관적으로 코딩할 수 있게 된다.

In [ ]:
print('''\
Hello
World
''')

Hello
World



**[4]**  
'Hello'  
(단, 작은 따옴표도 함께 출력한다.)

In [ ]:
print("'Hello'")

'Hello'


**[5]**  
"Hello World"  
(단, 큰따옴표도 함께 출력한다.)

In [ ]:
print('"Hello World"')

"Hello World"


**[6]**  
"!@#$%^&*()"  
(단, 큰따옴표도 함께 출력한다.)

In [ ]:
print('"!@#$%^&*()"')

"!@#$%^&*()"


**[7]**  
"C:\Download\hello.cpp"  
(단, 큰따옴표도 함께 출력한다.)

In [ ]:
print('"C:\Download\hello.cpp"')

"C:\Download\hello.cpp"


**[8]**  
┌┬┐  
├┼┤  
└┴┘  
출력하기

In [ ]:
print('''\
┌┬┐
├┼┤
└┴┘
''')

┌┬┐
├┼┤
└┴┘



---
## **2. 기초-입출력**
---


**[10]**  
정수형(int)으로 변수를 선언하고, 변수에 정수값을 저장한 후
변수에 저장되어 있는 값을 그대로 출력해보자.

In [ ]:
var = int(input())
print(var)

13
13


**[11]**  
문자형(char)으로 변수를 하나 선언하고, 변수에 문자를 저장한 후
변수에 저장되어 있는 문자를 그대로 출력해보자.  

**Tip::**  
1. input()의 반환값은 기본으로 문자열로 정의된다.

In [ ]:
var = input()
print(var)

Hello
Hello


**[12]**  
실수형(float)로 변수를 선언하고 그 변수에 실수값을 저장한 후
저장되어 있는 실수값을 출력해보자.

In [ ]:
var = float(input())
print(var)

4
4.0


**[13]**  
정수(int) 2개를 입력받아 그대로 출력해보자.
(단, 띄어쓰기를 기준으로 입력한다.)  
입력 : 1 5  
출력 : 1 5

**Tip::**  
1. 문자열의 메소드(함수)인 split()을 이용하면 문자열을 공백 기준으로 배열(iterable)로 만들어준다.
2. 매핑함수인 map()을 이용하면 배열(iterable)의 모든 원소를 첫 번째 매개변수(parameter)로 변환할 수 있다. 정확히는 감싸준다는 표현이 맞다. 
 - *예 ) map(int, ['1', '2', '3'])  >>  [1,2,3]*
3. 매핑함수 map()의 반환값은 map객체이다. 따라서 육안으로 확인하기 위해서는 list()로 변환시켜줘야한다.

In [ ]:
var = list(map(int, input().split()))
print(var[0], var[1])

1 5
1 5


**[14]**  
2개의 문자(ASCII CODE)를 입력받아서 순서를 바꿔 출력해보자.

**아스키 코드란?**
- 컴퓨터가 문자를 읽을 수 있도록 문자에 대응하는 숫자들이 존재한다.
- 예 ) A => 1100001
- 이때의 문자가 '아스키 문자'이며, 숫자가 '아스키 코드'이다.



In [ ]:
var = input().split()
print(var[1], var[0])

a B
B a


**[15]**  
실수(float) 1개를 입력받아 저장한 후,
저장되어 있는 값을 소수점 셋 째 자리에서 반올림하여
소수점 이하 둘 째 자리까지 출력하시오.  

**Tip::**  
1. 반올림 함수 round()를 이용하면 된다.

In [ ]:
var = round(float(input()), 2)
print(var)

3.1111
3.11


**[16]**  
int형 정수 1개를 입력받아 공백을 사이에 두고 3번 출력해보자.

In [ ]:
var = int(input())
print(var, var, var)

55
55 55 55


**[17]**  
어떤 형식에 맞추어 시간이 입력될 때, 그대로 출력하는 연습을 해보자.  
콜론(:) 기호를 기준으로 두 수가 각 변수에 저장된다.  

입력 : 3:15  
출력 : 3:15  

**Tip::**  
1. split()의 매개변수로 문자열을 분할하기 위한 기준을 정의할 수 있다.
2. 문자열의 메소드(함수)인 format()을 이용하면 문자열 내부에 변수값을 대입할 수 있다.



In [ ]:
h, m = input().split(':')
print('{}:{}'.format(h, m))

3:15
3:15


**[18]**  
년, 월, 일을 입력받아 지정된 형식으로 출력하는 연습을 해보자.  

**입력**  
연, 월, 일이 ".(닷)"으로 구분되어 입력된다.  
**출력**  
입력받은 연, 월, 일을 yyyy.mm.dd 형식으로 출력한다.

입력 : 2020.2.9  
출력 : 2020.02.09  
(단, m 혹은 d가 한 자리 수인 경우 앞에 0을 붙여 출력한다.)


In [ ]:
y, m, d = input().split('.')
if len(m) == 1:
  m = '0'+m
if len(d) == 1:
  d = '0'+d
print('{}.{}.{}'.format(y, m, d))

2020.2.9
2020.02.09


**[19]**  
주민번호는 다음과 같이 구성된다.  

XXXXXX-XXXXXXX  

앞의 6자리는 생년월일(yymmdd)이고 뒤 7자리는 성별, 지역, 오류검출코드이다.  
주민번호를 입력받아 형태를 바꿔 출력해보자.  


**입력**  
주민번호 앞 6자리와 뒷 7자리가 '-'로 구분되어 입력된다.
(입력값은 가상의 주민번호이다.)
ex)110011-0000000  
**출력**  
'-'를 제외한 주민번호 13자리를 모두 붙여 출력한다.

입력 : 000907-1121112  
출력 : 0009071121112

In [ ]:
a, b = input().split('-')
print('{}{}'.format(a,b))

000907-1121112
0009071121112


**[20]**  
1개의 문자열을 입력받아 그대로 출력해보자.

In [ ]:
string = input()
print(string)

우리밋 최고에요
우리밋 최고에요


**[22]**  
공백이 포함되어 있는 한 문장이 입력된다.
단, 입력되는 문장은 여러 개의 단어로 구성되고, 엔터로 끝난다.

In [ ]:
string = input()
print(string)

programing is very good
programing is very good


**[23]**  
실수 1개를 입력받아 정수 부분과 실수 부분으로 나누어 출력한다.

입력 :   
1.435867  

출력 :  
1  
435867

In [ ]:
string = input().split('.')
print('''\
{}
{}
'''.format(string[0], string[1]))

1.435867
1
435867



**[24]**  
단어를 1개 입력받는다.  
입력받은 단어(영어)의 각 문자를 한줄에 한 문자씩 분리해 출력한다.  
(단, 단어의 문자(영어)를 하나씩 나누어 한 줄에 한 개씩 ' '로 묶어서 출력한다.)  

입력 :  
'Boy'

출력 :  
'B'  
'o'  
'y'  

**Tip::**  
1. str도 List와 동일하게 배열과 같은 형식으로 접근가능하다. 문자열도 리스트와 같이 iterable 객체이기 때문이다. 
 - *ex) '문자열'[0] >> '문'*  
2. 반복문 for()를 이용하여 문자열의 길이만큼 반복한다.

In [ ]:
string = input()
for i in range(len(string)):
  print("'{}'".format(string[i]))

WooRim
'W'
'o'
'o'
'R'
'i'
'm'


**[25]**  
다섯 자리의 정수 1개를 입력받아 각 자리별로 나누어 출력한다.  

입력 :  
75254

출력 :  
[70000]  
[5000]  
[200]  
[50]  
[4]  

**Tip::**  
1. 문자열 연산이 가능함을 잊지 말자.
 - ex) '문자'*5 >> '문자문자문자문자문자'


In [ ]:
integer = input()
count = len(integer)-1
for i in range(len(integer)):
  print([int(integer[i] + '0'*count)])
  count -= 1

75254
[70000]
[5000]
[200]
[50]
[4]


**[26]**  
입력되는 시:분:초 에서 분만 출력해보자.

In [ ]:
h, m, s = input().split(':')
print(m)

17:24:55
24


**[27]**  
년월일을 출력하는 방법은 나라마다, 형식마다 조금씩 다르다.

년월일(yyyy.mm.dd)를 입력받아,

일월년(dd-mm-yyyy)로 출력해보자.

(단, 한 자리 일/월은 0을 붙여 두자리로 출력한다.) 

**Tip::**  
1. 조건문 if-else문을 파이썬의 3항 연산자(Chapter 10 참고) 기능을 이용하면 더 간단하게 작성할 수 있다. 이렇게 작성하는 것이 메모리 효율성면에서도 효과적이다. (18번 참고)



In [ ]:
y, m, d = input().split('.')
m = '0'+m if len(m) == 1 else m
d = '0'+d if len(d) == 1 else d
print('{}-{}-{}'.format(d, m, y))

2020.2.11
11-02-2020




---


##**3. 기초-데이터형**


---



**[28]**  
정수 1개를 입력받아 그대로 출력해보자.
```
파이썬에서의 int() 데이터형의 크기는 4바이트(32비트)로 지정되어있다.
이보다 큰 범위를 지정하고자 할 때는 long 데이터 형을 이용하면 된다.
파이썬에서는 4가지의 데이터형을 제공한다.
```
**파이썬에서 제공하는 데이터형**  
int (plain integers) : 정수  
long (long integers) : int 보다 범위가 큰 정수(메모리 한계까지 저장 가능, 사실상 무제한)  
float (floating point numbers) : 실수  
complex (complex numbers) : 복소수  

In [ ]:
integer = int(input())
print(integer)

500
500


**[29]**  
실수 1개를 입력받아 그대로 출력해보자.

In [ ]:
fl = float(input())
print(fl)

40
40.0


**[30]**  
정수 1개를 입력받아 그대로 출력해보자.



In [ ]:
integer = int(input())
print(integer)

-234234818
-234234818




---


## **4. 기초-출력변환**

---



**[31]**

10진수를 입력받아 8진수(octal)로 출력해보자.

**Tip::**  
1. 8진수는 '0o12'처럼 숫자 앞에 '0o'를 붙여준다. 이것으로 '12'가 8진수의 '12'임을 인지한다.
2. 10진수를 8진수로 변환하는 함수는 oct()이며, '0o12'와 같이 출력되므로 앞 두자리를 생략하고 출력하면된다. 따라서 '0o12'[2:]를 해주면 되는 것.

In [ ]:
octal = int(input())
print( oct(octal)[2:] )

10
12


**[32]**  

10진수를 입력받아 16진수(hexadecimal)로 출력해보자.  

**Tip::**  
1. 16진수는 '0xa'처럼 숫자 앞에 '0x'를 붙여준다. 이것으로 'a'가 16진수의 'a'임을 인지한다.
2. 10진수를 16진수로 변환하는 함수는 hex()이며, '0xa'와 같이 출력되므로 앞 두자리를 생략하고 출력하면된다. 따라서 '0xa'[2:]를 해주면 되는 것.

In [ ]:
hexadecimal = int(input())
print( hex(hexadecimal)[2:] )

10
a


**[33]**  
10진수를 입력받아 16진수(hexadecimal)로 출력해보자.  
16진수(**대문자**)로 출력한다.  


**Tip::**  
1. 소문자를 대문자로 변환하려면 문자열의 메소드(함수)인 upper()를 이용하면 된다.

In [ ]:
hexadecimal = int(input())
hexConv = hex(hexadecimal)[2:]
print( hexConv.upper() )

10
A


**[34]**  
8진수로 입력된 정수 1개를 10진수로 바꾸어 출력해보자.

**Tip::**  
1. 10진수로 변환하고자 할 때는 int()를 이용하면 된다. 첫번째 파라미터로는 변환하고자하는 숫자의 문자열('0o12')을, 두번째 파라미터로는 첫번째 값이 몇 진수인지를 정수로 입력하면 된다.

In [ ]:
octal = '0o' + input()
print( int(octal, 8) )


12
10


**[35]**  

16진수로 입력된 정수 1개를 **8진수**로 바꾸어 출력해보자.

**Tip::**  
1. 16진수 >> 10진수 >> 8진수 순서대로 변환

In [ ]:
hexadecimal = '0x' + input()
integer = int(hexadecimal, 16)
print( oct(integer)[2:] )


10
20


**[36]**

영문자 1개를 입력받아 아스키 코드표의 10진수 값으로 출력해보자.

**아스키 코드란?**
- 문자를 이진수의 7비트로 표현한 것
- 예 ) A => 1100001

**Tip::**  
1. 문자열에 대응되는 아스키코드를 반환해주는 함수 ord()를 이용하면 된다.

In [ ]:
askii = ord(input())
print( askii )

a
97


**[37]**

10진 정수 1개를 입력받아 아스키 문자로 출력해보자.


In [ ]:
askii = chr(int(input()))
print( askii )

97
a




---


## **5. 기초-산술연산**


---



**[38]**  
정수 2개를 입력받아 합을 출력하는 프로그램을 작성해보자.


In [ ]:
a, b = map(int, input().split())
print( a+b )

100 -101
-1


**[39]**  
정수 2개를 입력받아 합을 출력하는 프로그램을 작성해보자.
```
실제 문제에서는 굉장히 넓은 정수 범위의 데이터형을 요구하는 문제이나 파이썬에서는 int()로 처리가 가능하다.
예를 들어 C에서 unsinged int보다도 크며 unsinged long과 같은 범위를 지니고 있다고 한다.
```

**실제 범위**

범위 : -9223372036854775808 ~ 9223372036854775807



```
이보다 큰 범위를 지정하고자 할 때는 long 데이터 형을 이용하면 된다.
파이썬에서는 4가지의 데이터형을 제공한다.
```
**파이썬에서 제공하는 데이터형**  
int (plain integers) : 정수  
long (long integers) : int 보다 범위가 큰 정수  
float (floating point numbers) : 실수  
complex (complex numbers) : 복소수  

In [ ]:
a, b = map(int, input().split())
print( a+b )

-394792 -394792
-789584


**[40]**

입력된 정수의 부호를 바꿔 출력해보자.


In [ ]:
intConv = -int(input())
print( intConv )

-20498
20498


**[41]**

영문자 1개를 입력받아 그 다음 문자를 출력해보자.

영문자 'A'의 다음 문자는 'B'이고, 영문자 '0'의 다음 문자는 '1'이다.

**Tip::**  
1. 아스키 코드를 이용하면 된다.
2. 'A' == 97(1000001), 'B' == 98(1000010) 이므로 아스키 코드로 변환된 숫자에 1을 더한 뒤 아스키 문자로 재변환 해주면 된다.

In [ ]:
order = ord(input())
print( chr(order+1) )

A
B


**[42]**

정수 2개(a, b) 를 입력받아 a를 b로 나눈 몫을 출력해보자.

**Tip::**  
1. 산술 연산자 '/'는 정확히 나눈 후의 값을 반환한다. *ex) 1/3 >> 0.33333*
2. 산술 연산자 '//'는 나눈 후의 몫만 반환한다. *ex) 1/3 >> 0*



In [ ]:
a, b = map(int, input().split())
print( a//b )

1 3
0


**[43]**

정수 2개(a, b) 를 입력받아 a를 b로 나눈 나머지를 출력해보자.

In [ ]:
a, b = map(int, input().split())
print( a%b )

1 3
1


**[44]**

정수를 1개 입력받아 1만큼 더해 출력해보자.


In [ ]:
a = int(input())
print( a+1 )

155
156


**[45]**

정수 2개(a, b)를 입력받아 합, 차, 곱, 몫, 나머지, 나눈 값을 자동으로 계산해보자.
```
첫 줄에 합  
둘째 줄에 차,  
셋째 줄에 곱,  
넷째 줄에 몫,  
다섯째 줄에 나머지,  
여섯째 줄에 나눈 값을 순서대로 출력한다.  
(실수, 소수점 이하 셋째 자리에서 반올림해 둘째 자리까지 출력)  
```



In [ ]:
a, b = map(int, input().split())
print( a+b )
print( a-b )
print( a*b )
print( a//b )
print( a%b )
print( round(a/b, 2) )

11 7
18
4
77
1
4
1.57


**[46]**

정수 3개를 입력받아 합과 평균을 출력해보자.
```
합과 평균을 줄을 바꿔 출력한다.
평균은 소수점 이하 둘째 자리에서 반올림해서 소수점 이하 첫째 자리까지 출력한다.
```



In [ ]:
a, b, c = map(int, input().split())
print( a+b+c )
print( round((a+b+c)/3, 1) )

10 15 25
50
16.7




---


## **6. 기초-비트 시프트 연산**


---



**[47]**

정수 1개를 입력받아 2배 곱해 출력해보자.

**Tip::**  
1. 곱하기 2를 해도 되지만 이진수를 왼쪽으로 한 번씩 이동시켜주면 기존 값의 두 배가 된다.
 - *ex) 5(101) => 10(1010), 7(111) => 14(1110)*

In [ ]:
mul = int(input())
print( mul<<1 )

5
10


**[48]**

정수 2개(a, b)를 입력받아 a를 2(b 제곱)배 곱한 값으로 출력해보자.
( a * 2(b 제곱) )

**Tip::**  
![팁](https://www.codeup.kr/upload/pimg6164_1.png)

In [ ]:
a, b = map(int,input().split())
print( a<<b )

12 3
96




---


## **7. 기초-비교연산**


---




**[49]**

두 정수(a, b)를 입력받아 a가 b보다 크면 1을, a가 b보다 작거나 같으면 0을 출력하는 프로그램을 작성해보자.

```
**참고**
어떤 값을 비교하기 위해 비교/관계연산자(comparison/relational)를 사용할 수 있다.

비교/관계연산자 > 는
왼쪽의 값이 오른쪽 값 보다 큰 경우 참(true)을 나타내는 정수값 1로 계산하고,
그 외의 경우에는 거짓(false)를 나타내는 정수값 0으로 계산한다.

비교/관계연산자도 일반적인 사칙연산자처럼 주어진 두 수를 이용해 계산을 수행하고,
그 결과를 1(참), 또는 0(거짓)으로 계산해 주는 연산자이다.

비교/관계연산자는 >, <, >=, <=, ==(같다), !=(다르다) 6개가 있다.
```

**Tip::**  

![대체 텍스트](https://www.codeup.kr/upload/pimg6165_1.png)

1. 조건문 if-elif를 사용하여 조건들 마다의 실행 구문을 제시해줄 수 있다.


In [ ]:
a, b = map(int, input().split())
if a > b:
  print( 1 )
elif a <= b:
  print( 0 )

5 7
0
0
0


**[50]**

두 정수(a, b)를 입력받아
a와 b가 같으면 1을, 같지 않으면 0을 출력하는 프로그램을 작성해보자.

In [ ]:
a, b = map(int, input().split())
if a == b:
  print( 1 )
elif a != b:
  print( 0 )

7 5
0


**[51]**

두 정수(a, b)를 입력받아
b가 a보다 크거나 같으면 1을, 그렇지 않으면 0을 출력하는 프로그램을 작성해보자.


**Tip::**  
1. "그렇지 않으면" 이라는 조건은 else로 처리할 수 있다.

In [ ]:
a, b = map(int, input().split())
if b >= a:
  print( 1 )
else:
  print( 0 )

234 57532
1


**[52]**

두 정수(a, b)를 입력받아
a와 b가 서로 다르면 1을, 그렇지 않으면 0을 출력하는 프로그램을 작성해보자.


In [ ]:
a, b = map(int, input().split())
if b != a:
  print( 1 )
else:
  print( 0 )

10 13
1




---


## **8. 기초-논리연산**
---

**[53]**

1(true, 참) 또는 0(false, 거짓) 이 입력되었을 때
반대로 출력하는 프로그램을 작성해보자.

```
**참고**
파이썬에서 비교/관계 연산(==, !=, >, <, >=, <=)이 수행될 때,
0은 거짓(false)으로 인식되고, 0이 아닌 모든 수는 참(true)으로 인식된다.

참 또는 거짓의 논리값을 역(반대)으로 바꾸기 위해서는
논리(logical)연산자 'not'를 사용할 수 있다.
이러한 논리연산을 NOT 연산이라고 부른다.

참, 거짓의 논리값(boolean value)인 불 값을 다루어주는 논리연산자는
'not', 'and', 'or'가 있다.

** 불 대수(boolean algebra)는 수학자 불이 만들어낸 것으로
참/거짓만 가지는 논리값과 그 연산을 다룬다.
```

**Tip::**  
1. 참 또는 거짓의 논리값을 역(반대)으로 바꾸기 위해서는
논리(logical)연산자 'not'를 사용할 수 있다.

![대체 텍스트](https://www.codeup.kr/upload/pimg6169_1.png)


In [ ]:
boolean = int(input())
print( not boolean )

True


**[54]**

두 개의 참(1) 또는 거짓(0)이 입력될 때,
모두 참일 때에만 참을 출력하는 프로그램을 작성해보자.

**Tip::**  
1. 파이썬에서는 AND연산값이 참이면 뒤에 있는 값을 출력하게 된다.
 - ex) '앞' and '뒤' >> '뒤'

 ![대체 텍스트](https://www.codeup.kr/upload/pimg6170_1.png)

In [ ]:
for _ in range(4):
  a, b = map(int, input().split())
  print( a and b )

0 131
0
131 456
456
777 888
888
888 0
0


**[55]**

두 개의 참(1) 또는 거짓(0)이 입력될 때,
하나라도 참이면 참을 출력하는 프로그램을 작성해보자.

**Tip::**  
1. 파이썬에서는 OR연산값이 참이면 참인 값을 출력하게 된다.
 - ex) '참' or 0 >> '참
 - ex) 0 or True >> True

 ![대체 텍스트](https://www.codeup.kr/upload/pimg6171_1.png)

In [ ]:
for i in range(4):
  a, b = map(int, input().split())
  print( a or b )

0 555
555
678 0
678
333 2212
333
111 222
111


**[우리밋이 알려주는 Bonus 문제 (1)]**

1개의 정수형 입력이 들어오면 논리 연산을 이용하여 '홀수'와 '짝수'를 판별하여라

**Tip::**  

1. 입력이 2로 나눠지면 짝수이고, 그렇지 않으면 홀수임을 이용한다.
2. 파이썬에서의 AND와 OR의 특징을 이용한다. => 파이썬에서는 이러한 연산을 "삼항 연산"으로 정의한다.(삼항연산은 Chapter 10에서 다룬다)

In [ ]:
number = int(input())
print( number%2 and '홀수' or '짝수' )

334
짝수


**[56]**

두 가지의 참(1) 또는 거짓(0)이 입력될 때,
참/거짓이 서로 다를 때에만 참을 출력하는 프로그램을 작성해보자.


**Tip::**  
1. 이러한 논리연산을 XOR(exclusive or, 배타적 논리합)연산이라고도 부른다. 이를 표현하기 위해서는 (a AND (NOT b)) OR ((NOT a) AND b)처럼 하면 된다.  
![대체 텍스트](https://www.codeup.kr/upload/pimg6172_1.png)

In [ ]:
for _ in range(4):
  a, b = map(int, input().split())
  print( (a and (not b)) or ((not a) and b) )

1 1
False
1 0
True
0 1
1
0 0
0


**[57]**

두 개의 참(1) 또는 거짓(0)이 입력될 때,
참/거짓이 서로 같을 때에만 참이 계산되는 프로그램을 작성해보자.

**Tip::**  
![대체 텍스트](https://www.codeup.kr/upload/pimg6173_1.png)

In [ ]:
for _ in range(4):
  a, b = map(int, input().split())
  print( ((not a) and (not b)) or (a and b) )

0 0
True
0 1
0
1 0
0
1 1
1


**[58]**

두 개의 참(1) 또는 거짓(0)이 입력될 때,
모두 거짓일 때에만 참이 계산되는 프로그램을 작성해보자.

**Tip::**  
![대체 텍스트](https://www.codeup.kr/upload/pimg6174_1.png)

In [ ]:
for _ in range(4):
  a, b = map(int, input().split())
  print( not(a or b) )

0 0
True
0 1
False
1 0
False
1 1
False






---


## **9. 기초-비트단위 논리연산**


---



**[59]**

입력 된 정수를 **비트단위**로 참/거짓을 바꾼 후 정수로 출력해보자.  

예를 들어 1이 입력되었을 때 저장되는  
**1을 32비트 2진수로 표현하면 00000000 00000000 00000000 00000001 이고,  
~1은 11111111 11111111 11111111 11111110 가 되는데 이는 -2를 의미한다.**  


**Tip::**  

비트단위(bitwise)연산자 ' ~ ' 를 붙이면 된다.  
(~ : tilde, 틸드라고 읽는다.)  
```
**비트단위(bitwise) 연산자**
~(bitwise not),  
&(bitwise and),  
|(bitwise or),  
^(bitwise xor),
<<(bitwise left shift),  
>>(bitwise right shift)
```

**틈새 자료구조**(오버플로우 설명)
```
**참고**
컴퓨터에 저장되는 모든 데이터들은 2진수 형태로 바뀌어 저장된다.
0과 1로만 구성되는 비트단위들로 변환되어 저장되는데,
양의 정수는 2진수 형태로 바뀌어 저장되고,
음의 정수는 "2의 보수 표현"방법으로 저장된다.

예를 들어 파이썬에서는 int형(4바이트(byte), 32비트)으로 선언된 변수에 양의 정수 5를 저장하면
5의 2진수 형태인 101이 32비트로 만들어져
00000000 00000000 00000000 00000101
로 저장된다.(공백은 보기 편하도록 임의로 분리)

int 형의 정수 0은
00000000 00000000 00000000 00000000

그리고 -1은 0에서 1을 더 빼고 32비트만 표시하는 형태로
11111111 11111111 11111111 11111111 로 저장된다.

-2는 -1에서 1을 더 빼면 된다.
11111111 11111111 11111111 11111110 로 저장된다.

그래서 int 형으로 선언된 변수에는 최소 -2147483648 을 의미하는
10000000 00000000 00000000 00000000 부터

최대 +2147483647 을 의미하는
01111111 11111111 11111111 11111111 로 저장될 수 있는 것이다.

그렇다면 -2147483648
10000000 00000000 00000000 00000000 에서 1을 더 뺀다면?

01111111 11111111 11111111 11111111 이 된다.
즉 -2147483649 가 아닌 +2147483647 이 되는 것이다.

이러한 것을 "오버플로우(overflow, 넘침)"라고 한다.

이러한 내용을 간단히 표시하면, 정수 n이라고 할 때,

~n = -n - 1
-n = ~n + 1 과 같은 관계로 표현된다.
```



In [ ]:
bitNot = ~int(input())
print( bitNot )

5
-6


**[60]**

입력된 정수 두 개를 **비트단위**로 and 연산한 후 그 결과를 정수로 출력해보자.

예를 들어 3과 5가 입력되었을 때를 살펴보면  
3       : 00000000 00000000 00000000 00000011  
5       : 00000000 00000000 00000000 00000101  
3 & 5 : 00000000 00000000 00000000 00000001  
이 된다.


**Tip::**  
1. 비트단위(bitwise)연산자 &를 사용하면 된다.(and, ampersand, 앰퍼센드라고 읽는다.)
2. 비트단위 and 연산은 두 비트열이 주어졌을 때,
둘 다 1인 부분의 자리만 1로 만들어주는 것과 같다.


```
이 연산을 이용하면 어떤 비트열의 특정 부분만 모두 0으로도 만들 수 있는데
192.168.0.31   : 11000000.10101000.00000000.00011111
255.255.255.0 : 11111111.11111111.11111111.00000000

두 개의 ip 주소를 & 연산하면
192.168.0.0 : 110000000.10101000.0000000.00000000 을 계산할 수 있다.

실제로 이 계산은 네트워크에 연결되어 있는 두 개의 컴퓨터가 데이터를 주고받기 위해
같은 네트워크에 있는지 아닌지를 판단하는데 사용된다.

이러한 비트단위 연산은 빠른 계산이 필요한 그래픽처리에서
마스크연산(특정 부분을 가리고 출력하는)을 수행하는 데에도 효과적으로 사용된다
```



In [ ]:
a, b = map(int, input().split())
print( a&b )

3 5
1


**[우리밋이 알려주는 Bonus 문제 (2)]**

1개의 정수형 입력이 들어오면 비트 연산을 이용하여 '홀수'와 '짝수'를 판별하여라

**Tip::**  

1. '짝수'와 '홀수'를 리스트에 담고 짝수일 때는 '짝수'를, 홀수일 때는 '홀수'를 출력하게 한다.
2. 0이 아닌 어떠한 정수일지라도 1과 비트단위 논리곱(&)을 수행하게 되면 1이 되는 특성을 이용한다.


In [ ]:
number = int(input())
print( ['짝수', '홀수'][number & 1] )

100
짝수


**[61]**

입력된 정수 두 개를 비트단위로 or 연산한 후 그 결과를 정수로 출력해보자.

예를 들어 3과 5가 입력되었을 때를 살펴보면  
3     : 00000000 00000000 00000000 00000011  
5     : 00000000 00000000 00000000 00000101  
3 | 5 : 00000000 00000000 00000000 00000111  
이 된다.


**Tip::**  
1. 비트단위(bitwise) 연산자 |(or, vertical bar, 버티컬바)를 사용하면 된다.
2. | 은 파이프(pipe)연산자라고도 불리는 경우가 있다.
3. 비트단위 or 연산은 둘 중 하나라도 1인 자리를 1로 만들어주는 것과 같다.

이러한 비트단위 연산은 빠른 계산이 필요한 그래픽처리에서도 효과적으로 사용된다.



In [ ]:
a, b = map(int, input().split())
print( a|b )

3 5
7


**[62]**

입력된 정수 두 개를 비트단위로 xor 연산한 후 그 결과를 정수로 출력해보자.


예를 들어 3과 5가 입력되었을 때를 살펴보면  
3       : 00000000 00000000 00000000 00000011  
5       : 00000000 00000000 00000000 00000101  
3 ^ 5 : 00000000 00000000 00000000 00000110  
이 된다.


**Tip::**  
1. 비트단위(bitwise) 연산자 ^(xor, circumflex/caret, 서컴플렉스/카릿)를 사용하면 된다.
2. 주의 ^은 수학식에서 거듭제곱(power)을 나타내는 기호와 모양은 같지만,
파이썬에서는 전혀 다른 배타적 논리합(xor, 서로 다를 때 1)의 의미를 가진다.


![대체 텍스트](https://www.codeup.kr/upload/pimg6178_1.png)

이러한 비트단위 연산은 빠른 계산이 필요한 그래픽처리에서도 효과적으로 사용된다.

```
구체적으로 설명하자면,
두 장의 이미지가 겹쳐졌을 때 색이 서로 다른 부분만 처리할 수 있다.
배경이 되는 그림과 배경 위에서 움직이는 그림이 있을 때,
두 그림에서 차이만 골라내 배경 위에서 움직이는 그림의 색으로 바꿔주면
전체 그림을 구성하는 모든 점들의 색을 다시 계산해 입히지 않고
보다 효과적으로 그림을 처리할 수 있게 되는 것이다.
비행기 슈팅게임 등을 상상해보면 된다.
```





In [ ]:
a, b = map(int, input().split())
print( a^b )

138 255
117


**[우리밋이 알려주는 Bonus 문제 (2-2]**


```
해당 문제는 '카카오'와 '우아한 테크코스(이하 우테코)'의 실제 코딩 테스트 데모 문제였음을 알려드립니다.
카카오에서는 이 보다 훨씬 어려운 난이도의 코딩 문제들이 출제되며,
우테코에서는 우테코의 최고 난이도를 10으로 봤을 때 4~5 정도(지극히 주관적 평가임을 유의)에 해당합니다.

참고 : 필자는 해당 사에서 출제된 문제들을 풀어봤을 뿐 실제 코딩 시험은 치뤄본 적이 없습니다.
```


**문제**
직사각형을 만드는 데 필요한 4개의 점 중 3개의 좌표가 주어질 때, 나머지 한 점의 좌표를 구하려고 합니다. 점 3개의 좌표가 들어있는 배열 v가 매개변수로 주어질 때, 직사각형을 만드는 데 필요한 나머지 한 점의 좌표를 return 하도록 solution 함수를 완성해주세요.

단, 직사각형의 각 변은 x축, y축에 평행하며, 반드시 직사각형을 만들 수 있는 경우만 입력으로 주어집니다.

**제한사항**
1. v는 세 점의 좌표가 들어있는 2차원 배열입니다.
2. v의 각 원소는 점의 좌표를 나타내며, 좌표는 [x축 좌표, y축 좌표] 순으로 주어집니다.
3. 좌표 값은 1 이상 10억 이하의 자연수입니다.
4. 직사각형을 만드는 데 필요한 나머지 한 점의 좌표를 [x축 좌표, y축 좌표] 순으로 담아 return 해주세요.

**입력(1)**
```
[[1,4], [3,4], [3,10]]
```
**출력(1)**
```
[1,10]
```


**입력(2)**
```
[[1,1], [2,2], [1,2]]
```
**출력(2)**
```
[2,1]
```


**Tip::**  

1. 나머지 한 점의 좌표는 기존에 좌표에서 없는 좌표이니,  1번만 입력된 좌표들을 찾아서 반환해주면 된다.
2. XOR 연산을 통해 구현할 수도 있으며, 이 경우 코드의 실행 속도가 굉장히 빠르다.


In [ ]:
# 기본 풀이

coordinates = [[1,4], [3,4], [3,10]]

answer = []
for i in range(2):
  if coordinates[0][i] == coordinates[1][i]:
    answer.append(coordinates[2][i])
  elif coordinates[0][i] == coordinates[2][i]:
    answer.append(coordinates[1][i])
  elif coordinates[1][i] == coordinates[2][i]:
    answer.append(coordinates[0][i])

print( answer )

[1, 10]


In [ ]:
# XOR 적용

coordinates = [[1,4], [3,4], [3,10]]

result = []
result.append( coordinates[0][0] ^ coordinates[1][0] ^ coordinates[2][0] )
result.append( coordinates[0][1] ^ coordinates[1][1] ^ coordinates[2][1] )

print( result )

[1, 10]


---
## **10. 기초-삼항연산**
---

**[63]**

입력된 두 정수 a, b 중 큰 값을 출력하는 프로그램을 작성해보자.  
단, 조건문을 사용하지 않고 3항 연산자 'and or' 를 사용한다.

**Tip::**  
파이썬에서의 3항(세 개의 요소로 이루어지는) 연산자는 다른 언어와 사용 방식이 다소 다르다.  
*타 언어 : "조건식 ? (참일 때의 값) : (거짓일 때의 값)”*
1. 파이썬에서의 3항(ternary) 연산자는 "조건식 and (참일 때의 값) or (거짓일 때의 값)” 의 형태로 사용하는 연산자이다.
 - 조건식의 계산 결과가 참인 경우에는 'and' 오른쪽의 값 또는 식으로 바뀌고,
 - 거짓인 경우에는 'or' 오른쪽의 값 또는 식으로 바뀐다.
2. 이것이 가능한 이유는 파이썬이 인터프리터 방식으로 동작하기 때문이다.

```
**인터프리터**
파이썬은 스크립트 언어이며, 스크립트 언어는 대부분 인터프리터 방식으로 동작한다.
인터프리터는 컴파일러와 대조되는 의미이기에 컴파일과 함께 설명이 필요하다.

컴파일은 소스 코드를 처음부터 끝까지 모두 기게어로 번역한 후에 실행 파일을 만들고, 그것을 컴퓨터가 실행하는 방식이기 때문에 실행속도가 비교적 빠르다.
인터프리트는 소스 코드를 처음부터 끝까지 전체 번역을 하는 것이 아니라, 실행되는 즉시 그 순간의 코드만을 기계어로 번역해 나아간다. 
따라서 모든 소스 코드를 통으로 번역하는 컴파일에 비해 생산 속도는 빠르나, 모두 번역되기까지의 시간이 비교적 오래 걸리므로 실행 속도가 느린 것이다.

이러한 인터프리트(Interpret)를 실행하는 프로그램을 인터프리터라고 부르나, 실생활에서는 그다지 차이를 두고 얘기하진 않는다.
by. 우리밋
```



In [ ]:
a, b = map(int, input().split())
print( a>b and a or b )

거짓


**[64]**  
  
입력된 세 정수 a, b, c 중 가장 작은 값을 출력하는 프로그램을 작성해보자.
(단, 삼항 연산자 이용)

**Tip::**  
1. 파이썬에서는 논리연산을 이용한 삼항 연산자 외에 if-else를 이용한 삼항 연산도 제공한다. 이러한 이유는 논리연산을 이용한 삼항연산 때 발생할 수 있는 오류가 있기 때문이다.
 - 3항 논리 연산의 오류 : 5==5 and 5-5 or 5+5 >> 10
 - 위 연산에서는 5==5가 참이기 때문에 5-5가 실행되어 0이 출력되는 것이 맞다. 그러나 10이 출력된다. 이유는 논리 연산의 특성 때문이다.
 - 위 구문을 크게 따져보면 (True and false) or true 로 표현할 수 있다. 따라서 앞의 (True and false)가 false가 되므로 or 오른쪽에 true 값인 10이 출력된 것이다.
2. if-else를 이용한 삼항 연산은 "(참일 때의 값) if (조건식) else (거짓일 때의 값)"으로 나타낼 수 있다.
3. 삼항 연산자는 중첩하여 이용할 수 있다.

In [ ]:
a, b, c = map(int, input().split())
num = a if a<b else b
print(  num if num < c else c )


3 5 10
3


**[우리밋이 알려주는 Bonus 문제 (3)]**

1개의 정수형 입력이 들어오면 삼항 연산을 이용하여 '홀수'와 '짝수'를 판별하여라

**Tip::**  

1. 입력이 2로 나눠지면 짝수이고, 그렇지 않으면 홀수임을 이용한다.
2. 파이썬에서의 논리연산인 AND와 OR의 특징(Chapter 8 참고)을 이용한다. => 파이썬에서는 이러한 연산을 "삼항 연산"으로 정의한다.

**Tip2::**  
1. 입력이 2로 나눠지면 짝수이고, 그렇지 않으면 홀수임을 이용한다.
2. 파이썬에서의 삼항 연산은 if-else로도 구현할 수 있다.

In [ ]:
number = int(input())
print( '홀수' if number%2 else '짝수' )

121
홀수


---
## **11. 기초-조건/선택실행구조**
---

**[65]**

세 정수 a, b, c가 입력되었을 때, 짝수만 출력해보자.

**Tip1::**  
1. 조건문 이용
 - [조건문 배우러가기](https://blog.naver.com/dnfla420/221452997812)

**Tip2::**  
1. filter() 이용
 - [필터 함수 배우러가기](https://blog.naver.com/dnfla420/221749353805)
2. '*(asterisk)' 이용
 - ['*' Asterisk 배우러가기](https://blog.naver.com/dnfla420/221749146572)
3. 익명함수 lambda 이용
 - [익명함수 lambda 배우러가기](https://blog.naver.com/dnfla420/221750350240)

In [ ]:
# 1
a, b, c = map(int, input().split())
if not a%2: print(a)
if not b%2: print(b)
if not c%2: print(c)

#2
a, b, c = map(int, input().split())
print( *(filter(lambda num: num%2 == 0, [a, b, c])) )

4 6 8
4 6 8


**[66]**

세 정수 a, b, c가 입력되었을 때, 짝(even)/홀(odd)을 출력해보자.

[매핑함수 map() 배우러가기](https://blog.naver.com/dnfla420/221750001689)

In [ ]:
# 1
a, b, c = map(int, input().split())
print( 'odd' if a%2 else 'even' )
print( b%2 and 'odd' or 'even' )
print( ['even', 'odd'][c%2] )


#2
a, b, c = map(int, input().split())
print( *map(lambda num: 'odd' if num%2 else 'even', [a, b, c]) )

5 6 7
odd even odd


**[67]**

정수 1개가 입력되었을 때, 음(minus)/양(plus)과 짝(even)/홀(odd)을 출력해보자.

**입력**  
-4  
**출력**  
minus  
even


In [ ]:
num = int(input())
print( num>0 and 'plus' or 'minus' )
print( num%2 and 'odd' or 'even' )

-7
minus
odd


**[68]**

점수(정수, 0 ~ 100)를 입력받아 평가를 출력해보자.

```
**평가 기준**
점수 범위 : 평가
 90 ~  100 : A
 70 ~   89 : B
 40 ~   69 : C
  0 ~   39 : D
로 평가되어야 한다.
```

**Tip::**  
1. if-elif-else를 이용한다.


In [ ]:
grade = int(input())
if grade >= 90:
  print('A+')
elif grade >= 70:
  print('B')
elif grade >= 40:
  print('C')
else:
  print('D')

100
A+


**[69]**

평가를 문자(A, B, C, D, ...)로 입력받아 내용을 다르게 출력해보자.

**평가 내용**  
[평가 : 내용]  
A : best!!!  
B : good!!  
C : run!  
D : slowly~  
나머지 문자들 : what?  




In [ ]:
word = input()
if word == 'A':
  print('best!!!')
elif word == 'B':
  print('good!!')
elif word == 'C':
  print('run!')
elif word == 'D':
  print('slowly~')
else:
  print('what?')

A+
what?


**[70]**
월이 입력될 때 계절 이름이 출력되도록 해보자.

**예**  
[월 : 계절 이름]  
12, 1, 2 : winter  
  3, 4, 5  : spring  
  6, 7, 8  : summer  
  9, 10, 11 : fall  

In [ ]:
# 1
month = int(input())
if month==12 or month==1 or month==2:
  print('winter')
elif month==3 or month==4 or month==5:
  print('spring')
elif month==6 or month==7 or month==8:
  print('summer')
else:
  print('fall')

# 2
if month in [12,1,2]:
  print('winter')
elif month in [3,4,5]:
  print('spring')
elif month in [6,7,8]:
  print('summer')
else:
  print('fall')

3
spring


---
## **12. 기초-반복실행구조**
---

**[71]**  
정수가 순서대로 입력된다.  
(단, 개수는 알 수 없다.)

0이 아니면 입력된 정수를 출력하고, 0이 입력되면 출력을 중단해보자.  
while( ), for( ) 등의 반복문을 사용할 수 없다.

**입력**  
정수가 순서대로 입력된다.
```
7 4 2 3 0 1 5 6 9 10 8
```
**출력**  
입력된 정수를 줄을 바꿔 하나씩 출력하는데, 0이 입력되면 종료한다.
(0은 출력하지 않는다.)
```
7
4
2
3
```

**Tip::**  
원래 이 문제는 goto문을 이용해 풀도록 유도한 것인데 파이썬에서는 goto문을 제공하지 않으므로 함수를 이용하여 풀도록 한다.
1. 파이썬에서는 do~while()과 같은 반복문도 제공하지 않는다.
2. 파이썬에서는 함수를 이용하여 goto문처럼 사용하면 된다.  
 - [함수 배우러가기](https://wikidocs.net/24)

In [ ]:
# define : 정의하다
# 재귀함수

def goto(array, i):
  if array[i] == 0:
    return
  print(array[i])
  i += 1
  goto(array, i)

array = list(map(int, input().split()))
goto(array, i = 0)

7 4 2 3 0 1 5 6 9 10 8
7
4
2
3


**[72]**

n개의 정수가 순서대로 입력된다.  
(단 n의 최대 개수는 알 수 없다.)

n개의 입력된 정수를 순서대로 출력해보자.  
while( ), for( ) 등의 반복문을 사용할 수 없다.

**입력**  
첫 줄에 정수의 개수 n이 입력되고,
두 번째 줄에 n개의 정수가 공백을 두고 입력된다.
```
5  
1 2 3 4 5  
```
**출력**  
n개의 정수를 한 개씩 줄을 바꿔 출력한다.

```
1  
2  
3  
4  
5  
```

**Tip::**  
1. 리스트의 메소드인 reverse()는 내부 원소들을 거꾸로 재정렬해준다.



In [ ]:
n = int(input())
number = list(map(int, input().split()))

#1
print('== #1 ==')

def goto(number, n, i):
  if i == n: return
  print( number[i] )
  i += 1
  goto(number, n, i)

goto(number, n, 0)


#2 reverse 사용
print('== #2 ==')

number.reverse()
def goto(number, n):
  print(number[n])
  n -= 1
  if n == -1: return
  goto(number, n)

goto(number, n-1)

5
1 2 3 4 5
== #1 ==
1
2
3
4
5
== #2 ==
1
2
3
4
5


**[73]**

정수가 순서대로 입력된다.  
(단, 개수는 알 수 없다.)

0이 아니면 입력된 정수를 출력하고, 0이 입력되면 출력을 중단해보자.

**입력**  
정수가 순서대로 입력된다.
```
7 4 2 3 0 1 5 6 9 10 8
```
**출력**  
입력된 정수를 줄을 바꿔 하나씩 출력하는데, 0이 입력되면 종료한다.
(0은 출력하지 않는다.)
```
7
4
2
3
```

**Tip::**  
[반복문 for() 배우러 가기](https://blog.naver.com/dnfla420/221453090579)

In [ ]:
#1
number = map(int, input().split())

for element in number:
  if element is not 0:
    print(element)
    continue # continue를 만나면 아래의 구문은 실행하지 않고 다음 반복으로 넘어간다.
  break

3 4 5 2 6 3 0 4 2 5
3
4
5
2
6
3


In [ ]:
#2
number = map(int, input().split())

for value in number:
  if value == 0: break
  print( value )

3 4 5 2 6 3 0 4 2 5
3
4
5
2
6
3


**[74]**

정수(1 ~ 100) 1개가 입력되었을 때 카운트다운을 출력해보자.

**입력**  
정수 1개가 입력된다.
(1 ~ 100)
```
5
```
**출력**  
1씩 줄이면서 한 줄에 하나씩 1이 될 때까지 출력한다.
```
5
4
3
2
1
```

**Tip::**  
[반복문 for() 배우러 가기](https://blog.naver.com/dnfla420/221453090579)

In [ ]:
count = int(input())

for i in range(count, 0, -1):
  print( i )

10
10
9
8
7
6
5
4
3
2
1


**[75]**

정수(1 ~ 100) 1개가 입력되었을 때 카운트다운을 출력해보자.

**입력**  
정수 1개가 입력된다.
(1 ~ 100)
```
5
```
**출력**  
1씩 줄이면서 한 줄에 하나씩 1이 될 때까지 출력한다.
```
4
3
2
1
0
```

**Tip::**  
[반복문 for() 배우러 가기](https://blog.naver.com/dnfla420/221453090579)

In [ ]:
count = int(input())

for i in range(count-1, -1, -1):
  print( i )

10
9
8
7
6
5
4
3
2
1
0


**[76]**

영문자(a ~ z) 1개가 입력되었을 때 그 문자까지의 알파벳을 순서대로 출력해보자.

**입력**  
영문자 1개가 입력된다.
(a ~ z)
```
f
```
**출력**  
a부터 입력한 문자까지 순서대로 공백을 두고 출력한다.
```
a b c d e f
```

**Tip::**  
[반복문 for() 배우러 가기](https://blog.naver.com/dnfla420/221453090579)
1. 아스키 코드를 이용하면 된다. 
 - ord() : 문자를 아스키 코드의 10진수로 변환 *ex) 'a' >> 97*
 - chr() : 10진수의 아스키 코드를 문자로 변환 *ex) 97 >> 'a'*
2. print() 함수의 두번째 파라미터인 end는 줄바꿈을 없애줄 수 있다. default값이 줄바꿈이다.

In [ ]:
converter = ord(input())

for i in range(97, converter+1):
  print( chr(i), end=' ' )

h
a b c d e f g h 

**[77]**

정수(0 ~ 100) 1개를 입력받아 0부터 그 수까지 순서대로 출력해보자.

**입력**  
정수 1개가 입력된다.
(0 ~ 100)
```
4
```
**출력**  
0부터 그 수까지 줄을 바꿔 한 개씩 출력한다.
```
0
1
2
3
4
```

**Tip::**  
[반복문 while() 배우러 가기](https://blog.naver.com/dnfla420/221746836000)
1. for()로 푸는 것이 더 간단하나 while()문의 예를 보여드리기 위해 while로 풀었음.

In [ ]:
count = int(input())

i = 0
while count >= 0:
  print( i )
  i += 1
  count -= 1

15
0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15


In [ ]:
count = int(input())

for i in range(0, count+1):
  print( i )

15
0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15


---
## **13. 기초-종합**
---

**[78]** 짝수 합 구하기

정수(1 ~ 100) 1개를 입력받아 1부터 그 수까지 짝수의 합을 구해보자.

**입력**  
정수 1개가 입력된다.
(0 ~ 100)
```
5
```
**출력**  
1부터 입력된 수까지 짝수의 합을 출력한다.
```
6
```

**Tip::**  
1. range()함수의 첫 인자를 2로하고, 세 번째 인자를 2로 하면 2에서 2씩 증가하기 때문에 모든 숫자가 짝수가 된다. 따라서 이를 모두 더해주면 되는 것.

In [ ]:
num = int(input())

print('== #1 ==')
#1
answer = 0
for i in range(2, num+1, 2):
  answer += i
print(answer)

print('== #2 ==')
#2
answer2 = [i for i in range(2, num+1, 2)]
print( sum(answer2) )

print('== #3 ==')
#3
answer3 = range(2, num+1, 2)
print( sum(answer3) )

10
== #1 ==
30
== #2 ==
30
== #3 ==
30


**[79]** 원하는 문자가 입력될 때까지 반복 출력하기

'q'가 입력될 때까지 입력한 문자를 계속 출력하는 프로그램을 작성해보자.

**입력**  
문자들이 1개씩 계속해서 입력된다.  
```
x b k d l q g a c
```
**출력**  
'q'가 입력될 때까지 입력된 문자를 줄을 바꿔 한 줄씩 출력한다.
```
x
b
k
d
l
q
```

**Tip::**  
1. 입력값이 공백을 기준으로 한 번에 입력되기 때문에 input()을 매번 호출할 필요가 없다.
2. 입력된 문자열의 길이만큼만 반복하여 출력하면 되기에 for()문을 사용해도 된다.

In [ ]:
word = input().split()

print('== #1 ==')
#1
for w in word:
  print(w)
  if w == 'q': break

print('== #2 ==')
#2
i = 0
while word[i] != 'q':
  print(word[i])
  i += 1
print(word[i])

x b k d l q g a c
== #1 ==
x
b
k
d
l
q
== #2 ==
x
b
k
d
l
q


**[80]** 언제까지 더해야 할까?

1, 2, 3 ... 을 계속 더해 나갈 때,
그 합이 입력한 정수(0 ~ 1000)보다 같거나 작을 때까지
계속 더하는 프로그램을 작성해보자.

즉, 1부터 n까지 정수를 계속 더한다고 할 때,
어디까지 더해야 입력한 수보다 같거나 커지는지 알아보고자 하는 문제이다.

![대체 텍스트](https://www.codeup.kr/upload/pimg6196_1.png)

**입력**  
정수 1개가 입력된다.
```
55
```
**출력**  
1, 2, 3, 4, 5 ... 를 순서대로 계속 더해 합을 만들어가다가,
입력된 정수와 같거나 커졌을 때, 마지막에 더한 정수를 출력한다.
```
10
```


In [ ]:
end_point = int(input())

print('== #1 ==')
#1
total = 0
i = 0
while total < end_point:
  i += 1
  total += i
print( i )

print('== #2 ==')
#2
total = 0
for i in range(1, end_point+1):
  total += i
  if total >= end_point:
    print( i )
    break

55
== #1 ==
10
== #2 ==
10


**[81]** 주사위를 2개 던지면?

1부터 n까지, 1부터 m까지 숫자가 적힌
서로 다른 주사위 2개를 던졌을 때 나올 수 있는 모든 경우를 출력해보자.

**입력**  
주사위 2개의 면의 개수 n, m이 공백을 두고 입력된다.  
단, n, m은 10이하의 자연수
```
2 3
```
**출력**  
나올 수 있는 주사위의 숫자를 한 세트씩 줄을 바꿔 모두 출력한다.  
첫 번째 수는 n, 두 번째 수는 m으로 고정해 출력하도록 한다.
```
1 1
1 2
1 3
2 1
2 2
2 3
```
**Tip::**  
중첩의 원리...
1. 반복 실행 구조도 조건 실행 구조와 마찬가지로 중첩의 원리가 적용된다.
2. 반복 실행 구조를 중첩하면 원하는 반복 구조를 다양하게 만들어 낼 수 있다.

In [ ]:
n, m = map(int, input().split())

for n in range(1, n+1):
  for m in range(1, m+1):
    print(n, m)

2 3
1 1
1 2
1 3
2 1
2 2
2 3


**[82]** 16진수 구구단

16진수(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, F)를 배운
영일(01)이는 16진수끼리 곱하는 16진수 구구단에 대해서 궁금해졌다.

A, B, C, D, E, F 중 하나가 입력될 때,
1부터 F까지 곱한 16진수 구구단의 내용을 출력해보자.  
(단, A ~ F 까지만 입력된다.)

![대체 텍스트](https://www.codeup.kr/upload/pimg6198_1.png)

**입력**  
16진수로 한 자리 수가 입력된다.  
단, A ~ F 까지만 입력된다.
```
B
```
**출력**  
입력된 16진수에 1~F까지 순서대로 곱한, 16진수 구구단을 줄을 바꿔 출력한다.  
계산 결과도 16진수로 출력해야 한다.
```
B*1=B
B*2=16
B*3=21
B*4=2C
B*5=37
B*6=42
B*7=4D
B*8=58
B*9=63
B*A=6E
B*B=79
B*C=84
B*D=8F
B*E=9A
B*F=A5
```
**Tip::**  
1. 16진수 변환 함수 hex()를 이용해보자.
2. 소문자를 대문자로 변환하는 함수는 문자열의 메소드 중 upper()가 있다.

In [ ]:
char = input()

for i in range(1, 16):
  print( '%s*%s=%s' %(char, hex(i)[2:].upper(), hex(int(char, 16) * i)[2:].upper()) )

B
B*1=B
B*2=16
B*3=21
B*4=2C
B*5=37
B*6=42
B*7=4D
B*8=58
B*9=63
B*A=6E
B*B=79
B*C=84
B*D=8F
B*E=9A
B*F=A5


**[83]** 3 6 9 게임의 왕이 되자

3 6 9 게임을 하던 영일이는 3 6 9 게임에서 잦은 실수로 계속해서 벌칙을 받게 되었다.  
3 6 9 게임의 왕이 되기 위한 마스터 프로그램을 작성해 보자.
```
**3 6 9 게임은?**
여러 사람이 순서를 정해 순서대로 수를 부르는 게임이다.
만약 3, 6, 9 가 들어간 수를 자신이 불러야 하는 상황이면, 대신 "박수" 를 쳐야 한다.
33까지 진행했다면? "짝짝"과 같이 박수를 두 번 치는 형태도 있다.
```

**입력**  
10 보다 작은 정수 1개가 입력된다.
(1 ~ 9)
```
9
```
**출력**  
1 부터 그 수까지 순서대로 공백을 두고 수를 출력하는데,
3 또는 6 또는 9인 경우 그 수 대신 영문 대문자 X 를 출력한다.
```
1 2 X 4 5 X 7 8 X
```

**Tip::**  
1. 원래 369 게임은 숫자 중 3,6,9가 포함되어 있을 때 박수를 치는 게임이다.   
따라서 #2의 코드가 정답이어야 하지만 코드업에서는 3의 배수를 정답으로 처리하므로 #1과 같이 풀어야한다. 
  - 결론 : 3의 배수일 때 'X'가 출력되도록 해야한다.

In [ ]:
num = int(input())

print('== #1 정답 ==')
#1
for i in range(1, num+1):
  count = i if i%3 else 'X'
  print(count, end=' ')

print('\n== #2 오답 ==')
#2
for i in range(1, num+1):
  count = 'X' if '3' in str(i) or '6' in str(i) or '9' in str(i) else i
  print(count, end=' ')

36
== #1 정답 ==
1 2 X 4 5 X 7 8 X 10 11 X 13 14 X 16 17 X 19 20 X 22 23 X 25 26 X 28 29 X 31 32 X 34 35 X 
== #2 오답 ==
1 2 X 4 5 X 7 8 X 10 11 12 X 14 15 X 17 18 X 20 21 22 X 24 25 X 27 28 X X X X X X X X 

**[84]** 빛 섞어 색 만들기

빨강(red), 초록(green), 파랑(blue) 빛을 섞어
여러 가지 빛의 색을 만들어 내려고 한다.

빨강(r), 초록(g), 파랑(b) 각각의 빛의 개수가 주어질 때,  
(빛의 강약에 따라 0 ~ n-1 까지 n가지의 빛 색깔을 만들 수 있다.)

주어진 rgb 빛들을 다르게 섞어 만들 수 있는 모든 경우의 조합(r g b)과
총 가짓 수를 계산해보자.

**입력**  
빨녹파(r, g, b) 각 빛의 강약에 따른 가짓수(0 ~ 128))가 공백을 사이에 두고 입력된다.
예를 들어, 3 3 3 은 각 색깔 빛에 대해서 그 강약에 따라 0~2까지 3가지의 색이 있음을 의미한다.
```
2 2 2
```
**출력**  
만들 수 있는 rgb 색의 정보를 오름차순(계단을 올라가는 순, 12345... abcde..., 가나다라마...)으로
줄을 바꿔 모두 출력하고, 마지막에 그 개수를 출력한다.
```
0 0 0
0 0 1
0 1 0
0 1 1
1 0 0
1 0 1
1 1 0
1 1 1
8
```

**Tip::**  
![대체 텍스트](https://www.codeup.kr/upload/pimg6200_1.png)

In [ ]:
r, g, b = map(int, input().split())

count = 0
for i in range(r):
  for j in range(g):
    for k in range(b):
      print(i, j, k)
      count += 1
print(count)

2 2 2
0 0 0
0 0 1
0 1 0
0 1 1
1 0 0
1 0 1
1 1 0
1 1 1
8


**[85]** 소리 파일 저장용량 계산하기

```
**서론**
소리가 컴퓨터에 저장될 때에는 디지털 데이터화 되어 저장된다.

마이크를 통해 1초에 적게는 수십 번, 많게는 수만 번 소리의 강약을 체크해
그 값을 정수값으로 바꾸고, 그 값을 저장해 소리를 파일로 저장할 수 있다.

값을 저장할 때에는 비트를 사용하는 정도에 따라 세세한 녹음 정도를 결정할 수 있고,
좌우(스테레오) 채널로 저장하면 2배… 5.1채널이면 6배의 저장공간이 필요하고,
녹음 시간이 길면 그 만큼 더 많은 저장공간이 필요하다.
```
1초 동안 마이크로 소리강약을 체크하는 수를 h
(헤르쯔, Hz 는 1초에 몇 번? 체크하는가를 의미한다.)

한 번 체크한 결과를 저장하는 비트 b
(2비트를 사용하면 0 또는 1 두 가지, 16비트를 사용하면 65536가지..)

좌우 등 소리를 저장할 트랙 개수인 채널 c
(모노는 1개, 스테레오는 2개의 트랙으로 저장함을 의미한다.)

녹음할 시간 s가 주어질 때, 필요한 저장 용량을 계산하는 프로그램을 작성해보자.

```
**문제의 핵심 포인트**
실제로 일반적인 CD 음질(44.1KHz, 16bit, 스테레오)로 1초 동안 저장하려면
44100 * 16 * 2 * 1 bit의 저장공간이 필요
```
이렇게 녹음하는 방식을 PCM(Pulse Code Modulation) 방법이라고 하는데,
압축하지 않은 순수한(raw) 소리 데이터 파일은 대표적으로 *.wav 가 있다.
```
**틈새 자료구조**
8 bit(비트)           = 1byte(바이트)
1024 Byte(2^10 byte) = 1KB(킬로 바이트) 
1024 KB(2^10 KB)     = 1MB(메가 바이트)
1024 MB(2^10 MB)     = 1GB(기가 바이트)
1024 GB(2^10 GB)     = 1TB(테라 바이트)
```

**입력**  
h, b, c, s 가 공백을 두고 입력된다.   
h는 48,000이하, b는 32이하(단, 8의배수), c는 5이하, s는 6,000이하의 자연수이다.
```
44100 16 2 10
```
**출력**  
필요한 저장 공간을 MB 단위로 바꾸어 출력한다.   
단, 소수점 둘째 자리에서 반올림해 첫째 자리까지 출력하고 MB를 공백을 두고 출력한다.
```
1.7 MB
```

**Tip::**  
1. 음질의 1초 동안 저장 공간 / MB
> 'h' x 'b' x 'c' x 's' / ('8' x '1024' x '1024')

In [ ]:
h, b, c, s = map(int, input().split())

result = h*b*c*s
resultMB = result / (8 * 1024**2)
print(round(resultMB, 1), 'MB')

44100 16 2 10
1.7 MB


**[86]** 그림 파일 저장용량 계산하기

```
**서론**
이미지가 컴퓨터에 저장될 때에도 디지털 데이터화 되어 저장된다.

가장 기본적인 방법으로는 그림을 구성하는 한 점(pixel, 픽셀)의 색상을
빨강(r), 초록(g), 파랑(b)의 3가지의 빛의 세기 값으로 따로 변환하여 저장하는 것인데,

예를 들어 r, g, b 각 색에 대해서 8비트(0~255, 256가지 가능)씩을 사용한다고 하면,

한 점의 색상은 3가지 r, g, b의 8비트+8비트+8비트로 총 24비트로 표현해서
총 2^24 가지의 서로 다른 빛의 색깔을 사용할 수 있는 것이다.

그렇게 저장하는 점을 모아 하나의 큰 이미지를 저장할 수 있게 되는데,
1024 * 768 사이즈에 각 점에 대해 24비트로 저장하면 그 이미지를 저장하기 위한
저장 용량을 계산할 수 있다.
```
이미지의 가로 해상도 w, 세로 해상도 h, 한 픽셀을 저장하기 위한 비트 b 가 주어질 때,
압축하지 않고 저장하기 위해 필요한 저장 용량을 계산하는 프로그램을 작성해 보자.

```
**문제의 핵심 포인트**
실제로 일반적인 1024 * 768 사이즈(해상도)의 각점에 대해
24비트(rgb 각각 8비트씩 3개)로 저장하려면 1024 * 768 * 24 bit의 저장 용량이 필요하다.
```
이렇게 이미지의 원래(raw) 데이터를 압축하지 않고 그대로 저장하는 대표적인 이미지 파일이
*.bmp 파일이며, 비트로 그림을 구성한다고 하여 비트맵 방식 또는 래스터 방식이라고 한다.
```
**틈새 자료구조**
8 bit(비트)           = 1byte(바이트)
1024 Byte(2^10 byte) = 1KB(킬로 바이트) 
1024 KB(2^10 KB)     = 1MB(메가 바이트)
1024 MB(2^10 MB)     = 1GB(기가 바이트)
1024 GB(2^10 GB)     = 1TB(테라 바이트)
```

**입력**  
w, h, b 가 공백을 두고 입력된다.
단, w, h는 모두 정수이고 1~1024 이다. b는 40이하의 4의 배수이다.
```
1024 768 24
```
**출력**  
필요한 저장 공간을 MB 단위로 바꾸어 출력한다.
소수점 이하 셋째 자리에서 반올림해 둘째 자리까지 출력한 뒤 MB를 출력한다.

```
2.25 MB
```

**Tip::**  
1. 1024 * 768 사이즈(해상도)의 각점에 대해
24비트(rgb 각각 8비트씩 3개)로 저장하려면 1024 * 768 * 24 bit의 저장 용량이 필요하다.

In [ ]:
w, h, b = map(int, input().split())
result = (w*h*b) / (8 * 1024**2)
print( round(result, 2), 'MB' )

1024 768 24
2.25 MB


**[87]** 여기까지! 이제 그만~

1, 2, 3 ... 을 순서대로 계속 더해나갈 때,
그 합이 입력한 정수보다 작을 동안만 계속 더하는 프로그램을 작성해보자.

즉, 1부터 n까지 정수를 계속 더한다고 할 때,
어디까지 더해야 입력한 수보다 같거나 커지는지 알아보고자 하는 문제이다.

하지만, 이번에는 그 때의 합을 출력해야 한다.

예를 들어 57을 입력하면
1+2+3+...+8+9+10=55에 다시 11을 더해 66이 될 때,
그 값 66이 출력되어야 한다.

**입력**  
언제까지 합을 계산할 지, 정수 1개를 입력받는다.  
단, 입력되는 자연수는 100,000,000이하이다.
```
57
```
**출력**  
1, 2, 3, 4, 5 ... 순서대로 계속 더해가다가, 그 합이 입력된 정수보다 커지거나 같아지는 경우,
그때까지의 합을 출력한다.

```
66
```


In [ ]:
end_point = int(input())
total = 0
i = 1
while total < end_point:
  total += i
  i += 1
print( total )


57
66


**[88]** 3의 배수는 통과?

1부터 입력한 정수까지 1씩 증가시켜 출력하는 프로그램을 작성하되,
3의 배수인 경우는 출력하지 않도록 만들어보자.

예를 들면,
1 2 4 5 7 8 10 11 13 14 ...
와 같이 출력하는 것이다.

**입력**   
정수 1개를 입력받는다.  
(1 ~ 100)
```
10
```
**출력**  
1부터 입력한 정수보다 작거나 같을 때까지 1씩 증가시켜 출력하되
3의 배수는 출력하지 않는다.

```
1 2 4 5 7 8 10
```

**Tip::**

1. 반복문 안에서 continue;가 실행되면 그 아래의 내용을 건너뛰고, 다음 반복을 수행한다.
즉, 다음 반복으로 넘어가는 것이다.


In [ ]:
num = int(input())

for i in range(1, num+1):
  if i%3:
    print(i, end=' ')

15
1 2 4 5 7 8 10 11 13 14 

**[89]** 수 나열하기1

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열(series)이라고 한다.

예를 들어
1 4 7 10 13 16 19 22 25 ... 은
1부터 시작해 이전에 만든 수에 3을 더해 다음 수를 만든 수열이다.

이러한 것을 수학에서는 앞뒤 수들의 차이가 같다고 하여 
등차(차이가 같다의 한문 말) 수열이라고 한다.  
수열을 알게 된 영일이는 갑자기 궁금해졌다.

"그럼.... 123번째 나오는 수는 뭘까?"

영일이는 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.

시작 값(a), 등차(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.

**입력**   
시작 값(a), 등차의 값(d), 몇 번째 수 인지를 의미하는 정수(n)가
공백을 두고 입력된다.(모두 0 ~ 100)
```
1 3 5
```
**출력**  
n번째 수를 출력한다.

```
13
```

**Tip::**

1. 등차수열 리스트를 생성하고, 이것의 길이가 n값이 될 때까지 등차값을 추가한다.
2. 리스트의 마지막 값이 'n번째 수'가 되므로 마지막 값을 출력하면 된다. => '문자열'[-1] 을 하면 된다.


In [ ]:
a, d, n = map(int, input().split())

i = a
count = 0
arith = []
while count < n:
  arith.append(i)
  i += d
  count += 1
print( arith[-1] )

1 3 5
13


**[90]** 수 나열하기2

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열이라고 한다.

예를 들어
2 6 18 54 162 486 ... 은
2부터 시작해 이전에 만든 수에 3을 곱해 다음 수를 만든 수열이다.

이러한 것을 수학에서는 앞뒤 수들의 비율이 같다고 하여
등비(비율이 같다의 한문 말) 수열이라고 한다.


등비 수열을 알게된 영일이는 갑자기 궁금해졌다.

"그럼.... 13번째 나오는 수는 뭘까?"

영일이는 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.

시작 값(a), 등비(r), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.

**입력**   
시작 값(a), 등비의 값(r), 몇 번째 인지를 나타내는 정수(n)가
공백을 두고 입력된다.(모두 0 ~ 10)
```
2 3 7
```
**출력**  
n번째 수를 출력한다.

```
1458
```

**Tip::**

1. 등비수열 리스트를 생성하고, 이것의 길이가 n값이 될 때까지 등비값을 추가한다.
2. 리스트의 마지막 값이 'n번째 수'가 되므로 마지막 값을 출력하면 된다. => '문자열'[-1] 을 하면 된다.


In [ ]:
a, r, n = map(int, input().split())

i = a
count = 0
geom = []
while count < n:
  geom.append(i)
  i *= r
  count += 1
print( geom[-1] )

2 3 7
1458


**[91]** 수 나열하기3

어떤 규칙에 따라 수를 순서대로 나열한 것을 수열이라고 한다.

예를 들어
1 -1 3 -5 11 -21 43 ... 은
1부터 시작해 이전에 만든 수에 -2를 곱한 다음 1을 더해 다음 수를 만든 수열이다.

이런 이상한 수열을 알게 된 영일이는 또 궁금해졌다.

"그럼.... 13번째 나오는 수는 뭘까?"

영일이는 물론 수학을 아주 잘하지만 이런 문제는 본 적이 거의 없었다...

그래서 프로그램을 만들어 더 큰 수도 자동으로 계산하고 싶어졌다.


시작 값(a), 곱할 값(m), 더할 값(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때,
n번째 수를 출력하는 프로그램을 만들어보자.

**입력**   
시작 값(a), 곱할 값(m), 더할 값(d), 몇 번째 인지를 나타내는 정수(n)가
공백을 두고 입력된다.  
(a, m, d는 -50 ~ +50, n은 10이하의 자연수)  
```
1 -2 1 8
```
**출력**  
n번째 수를 출력한다.

```
-85
```

**Tip::**

1. 이상한 수열 리스트를 생성하고, 이것의 길이가 n값이 될 때까지 이상한(?)값을 추가한다.
2. 리스트의 마지막 값이 'n번째 수'가 되므로 마지막 값을 출력하면 된다. => '문자열'[-1] 을 하면 된다.


In [ ]:
a, m, d, n = map(int, input().split())

i = a
prog = []
while len(prog) < n:
  prog.append(i)
  i = i*m+d
print( prog[-1] )

1 -2 1 8
-85


**[92]** 함께 문제 푸는 날

같은 날 동시에 가입한 3명의 사람들이 온라인 채점시스템에 들어와 문제를 푸는 날짜가
매우 규칙적이라고 할 때, 다시 모두 함께 문제를 풀게 되는 그날은 언제일까?

예를 들어 3명이 같은 날 가입/등업하고, 각각 3일마다, 7일마다, 9일마다
한 번씩 들어온다면, 처음 가입하고 63일 만에 다시 3명이 함께 문제를 풀게 된다.

**입력**   
같은 날 동시에 가입한 인원 3명이 규칙적으로 방문하는,
방문 주기가 공백을 두고 입력된다. (단, 입력값은 100이하의 자연수이다.)
```
3 7 9
```
**출력**  
3명이 다시 모두 함께 방문해 문제를 풀어보는 날(동시 가입/등업 후 며칠 후?)을 출력한다.

```
63
```

**Tip::**

1. 세 방문 주기의 최소 공배수를 구해보자.


In [ ]:
a, b, c = map(int, input().split())
day = 1
while day%a != 0 or day%b != 0 or day%c != 0:
  day += 1
print( day )

day = 1
while 1:
  day += 1
  if day%a == 0 and day%b == 0 and day%c == 0: break
print( day )

3 7 9
63
63


---
## **14. 기초-1차원 배열**
---

###**[93]** 이상한 출석 번호 부르기1

정보 선생님은 수업을 시작하기 전에 이상한 출석을 부른다.

선생님은 출석부를 보고 번호를 부르는데,
학생들의 얼굴과 이름을 빨리 익히기 위해 번호를 무작위(랜덤)으로 부른다.

그리고 얼굴과 이름이 잘 기억되지 않는 학생들은 번호를 여러 번 불러
이름과 얼굴을 빨리 익히려고 하는 것이다.

출석 번호를 n번 무작위로 불렀을 때, 각 번호(1 ~ 23)가 불린 횟수를 각각 출력해보자.

**입력**   
첫 번째 줄에 출석 번호를 부른 횟수인 정수 n이 입력된다. (1 ~ 10000)  
두 번째 줄에는 무작위로 부른 n개의 번호(1 ~ 23)가 공백을 두고 순서대로 입력된다.
```
10
1 3 2 2 5 6 7 4 5 9
```
**출력**  
1번부터 번호가 불린 횟수를 순서대로 공백으로 구분하여 한 줄로 출력한다.

```
1 2 1 1 2 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

**Tip::**

1. 파이썬에서는 배열의 비어있는 공간을 미리 확보해 놓을 수 없다.따라서 필요한 갯수 만큼 '어떠한 값'으로 초기화 시켜주어야 한다. 
 - ex) LIST = list(range(23)) >> 0~22번지까지 0-22의 숫자들로 채워진다.
 - ex) LIST = [0 for _ in range(23)] >> 0~22번지까지 0으로 채워진다.
2. 호출되는 번호(주소값)의 값을 기존 값에서 +1하여 저장해주는 방식으로 해당 번호가 몇 번 호출되었는지 카운트할 수 있다.
3. 아래 코드에서 '*(Asterisk)'를 사용한 이유
 - 문제에서 원하는 출력값은 리스트 형태가 아닌, 내부의 원소를 그대로 출력하길 요구했기 때문이다.
 - ex) print( [1,2,3] ) >> [1,2,3]
 - ex) print( *[1,2,3] ) >> 1 2 3  
 - ['*(Asterisk)' 배우러가기](https://blog.naver.com/dnfla420/221749146572)


####정답

In [ ]:
n = int(input())
rand = map(int, input().split())
student = [0 for _ in range(23)]
for r in rand:
  student[r-1] += 1
print(*student)

10
1 3 2 2 5 6 7 4 5 9
1 2 1 1 2 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0


###**[94]** 이상한 출석 번호 부르기2



출석 번호를 n번 무작위로 불렀을 때, 부른 번호를 거꾸로 출력해 보자.

**입력**   
첫 번째 줄에 출석 번호를 부른 횟수인 정수 n이 입력된다. (1 ~ 10000)  
두 번째 줄에는 무작위로 부른 n개의 번호(1 ~ 23)가 공백을 두고 순서대로 입력된다.
```
10
10 4 2 3 6 6 7 9 8 5
```
**출력**  
출석을 부른 번호 순서를 바꾸어 공백을 두고 출력한다.
```
5 8 9 7 6 6 3 2 4 10
```

**Tip::**

1. 리스트의 메소드인 reverse()를 이용하면 된다.


#### 정답

In [ ]:
n = int(input())
rand = list(map(int, input().split()))
rand.reverse()
print(*rand)

10
10 4 2 3 6 6 7 9 8 5
5 8 9 7 6 6 3 2 4 10


###**[95]** 이상한 출석 번호 부르기3

출석 번호를 n번 무작위로 불렀을 때, 가장 빠른 번호를 출력해 보자.

**입력**   
첫 번째 줄에 출석 번호를 부른 횟수인 정수 n이 입력된다. (1 ~ 10000)  
두 번째 줄에는 무작위로 부른 n개의 번호(1 ~ 23)가 공백을 두고 순서대로 입력된다.
```
10
10 4 2 3 6 6 7 9 8 5
```
**출력**  
출석을 부른 번호 중에 가장 빠른 번호를 1개만 출력한다.
```
2
```

**Tip::**

1. 가장 작은 값을 찾으면 된다.
2. 파이썬에서는 배열(iterable) 객체의 원소 중 가장 작은 값을 반환해주는 min() 함수를 제공하고 있다.


####정답

In [ ]:
n = int(input())
rand = map(int, input().split())
print( min(rand) )

10
10 4 2 3 6 6 7 9 8 5
2


---
## **15. 기초-2차원 배열**
---

###**[우리밋의 LAST 보너스 문제]** 내 미래



```
해당 문제는 이차원 배열의 개념과 원리를 파악하기 위해 "우리밋"이 직접 만든 문제임을 알려드립니다. 
x축과 y축의 개념을 머릿 속에서 자유롭게 조작할 수 있도록 훈련하기 위해 만든 문제입니다.
이 점을 기억하시고 아래 문제를 푸시길 바랍니다.
해당 문제를 배포하시거나 외부에서 사용하실 때는 "우리밋"을 한번씩만 거론 부탁드리겠습니다.
구독만 해주신다면 그것으로 충분합니다.

ps. "내 미래"가 해당 문제 이름입니다. 참고로 전 다녀왔습니다. :)
```
훈련병인 철수는 교관의 지시에 따라야한다.  
교관은 "좌로 1보, 하로 2보 가!"와 같이 좌,우,상,하로 이동할 것을 명령한다.  
철수의 현재 위치가 입력으로 주어질 때 교관의 명령대로 이동한 위치는 어디일까?

```
제한 조건
1. 철수의 현재 위치는 첫 입력 값으로 공백을 두고 입력된다. 
  ex) 1 1 => (0, 0), 5 4 => (4, 3)
2. 훈련소의 전체 공간 크기는 5*5 이다.
3. 교관이 지시한 명령은 절대 훈련소 공간을 벗어나지 않는다.
4. 좌는 왼쪽, 우는 오른쪽, 상은 위쪽, 하는 아래쪽으로 한다.
5. 입력은 좌,우,상,하의 순서대로 공백을 두고 입력된다. 
  ex) 3 2 3 3 => 좌로 2보, 우로 2보, 상으로 3보, 하로 3보 이동.
```

**입력**
```
3 3
1 2 1 3
```
**출력**
```
0 0 0 0 0
0 0 0 0 0
0 0 0 0 0
0 0 0 0 0
0 0 0 1 0
```

**Tip::**
1. 좌표를 (i, j) 라고 볼 때 좌우로 이동은 j(가로축), 상하로 이동은 i(세로축)임을 기억하자.



####**정답**

In [ ]:
cur = tuple(map(int, input().split()))
lf, rgt, up, down = map(int, input().split())
coords = [[0 for _ in range(5)] for _ in range(5)]

move = ( cur[0] -up +down, cur[1] -lf +rgt )
coords[move[0] -1][move[1] -1] = 1

for crd in coords:
  print( *crd )

3 3
1 2 1 3
0 0 0 0 0
0 0 0 0 0
0 0 0 0 0
0 0 0 0 0
0 0 0 1 0


###**[96]** 바둑판에 흰 돌 놓기



영일이는 아버지와 함께 두던 매우 큰 오목에 대해서 생각해 보다가
"바둑판에 돌을 올린 것을 프로그래밍 할 수 있을까?"하고 생각하였다.

바둑판(19 * 19)에 n개의 흰 돌을 놓는다고 할 때,
n개의 흰 돌이 놓인 위치를 출력하는 프로그램을 작성해보자.

**입력**   
바둑판에 올려 놓을 흰 돌의 개수(n)가 첫 줄에 입력된다.  
둘째 줄 부터 n+1 번째 줄까지 흰 돌을 놓을 좌표(x, y)가 n줄 입력된다.  
n은 10이하의 자연수이고 x, y 좌표는 1 ~ 19 까지이며, 같은 좌표는 입력되지 않는다.
```
5
1 1
2 2
3 3
4 4
5 5
```
**출력**  
흰 돌이 올려진 바둑판의 상황을 출력한다.  
흰 돌이 있는 위치는 1, 없는 곳은 0으로 출력한다.
```
1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

**Tip::**

1. 파이썬에서는 2차원 배열(행렬)을 쉽게 다룰 수 있도록하는 numpy 라이브러리가 있다.
 - [Numpy 배우러가기](https://blog.naver.com/dnfla420/221752730611)
2. 문제에서 요구하는 방식대로 출력하기 위해 '*(Asterisk)'를 사용하였다.
 - ['*(Asterisk)' 배우러가기](https://blog.naver.com/dnfla420/221749146572)



####정답

In [ ]:
import numpy as np
n = int(input())
baduk = np.zeros((19,19), dtype=int)
# baduk = [[0 for _ in range(19)] for _ in rnage(19)]

for _ in range(n):
  x, y = map(int, input().split())
  baduk[x-1][y-1] = 1

for bd in baduk:
  print(*bd)

3
1 1
1 2
2 2
1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


###**[97]** 바둑판에 십자 뒤집기



부모님을 기다리던 영일이는 검정/흰 색 바둑알을 바둑판에 꽉 채워 깔아 놓고 놀다가...

"십(+)자 뒤집기를 해볼까?"하고 생각했다.

바둑판(19 * 19)에 흰 돌(1) 또는 검정 돌(0)이 모두 꽉 채워져 놓여있을 때,
n개의 좌표를 입력받아 십(+)자 뒤집기한 결과를 출력하는 프로그램을 작성해보자.


**입력**   
바둑알이 깔려 있는 상황이 19 * 19 크기의 정수값으로 입력된다.  
십자 뒤집기 횟수(n)가 입력된다.  
십자 뒤집기 좌표가 횟수(n) 만큼 입력된다. 단, n은 10이하의 자연수이다.
```
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
2
10 10
12 12
```
**출력**  
십자 뒤집기 결과를 출력한다.
```
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

**Tip::**

1. 파이썬에서는 2차원 배열(행렬)을 쉽게 다룰 수 있도록하는 numpy 라이브러리가 있다.
 - [Numpy 배우러가기](https://blog.naver.com/dnfla420/221752730611)
2. 문제에서 요구하는 방식대로 출력하기 위해 '*(Asterisk)'를 사용하였다.
 - ['*(Asterisk)' 배우러가기](https://blog.naver.com/dnfla420/221749146572)

**입력 예시 값 :**
```
baduk = [
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
]
```



####정답

In [ ]:
# 1. 초기 십자 바둑판 입력
baduk = []
for _ in range(19):
  matrix = list(map(int, input().split()))
  baduk.append(matrix)

# 2. n과 좌표 값을 입력받고 그에 따라 십자 형태에 맞춰 흑->백, 백->흑으로 변환
n = int(input())
for _ in range(n): # n번 입력
  x, y = map(int, input().split()) # x와 y 좌표 입력
  for i in range(19): # 변환
    baduk[i][y-1] = 1 if baduk[i][y-1] == 0 else 0 # 0->1, 1->0으로 변환
    baduk[x-1][i] = 1 if baduk[x-1][i] == 0 else 0 # 0->1, 1->0으로 변환

# 3. 변환된 값을 한 줄 단위로 출력
for i in range(19):
  print(*baduk[i])


2
10 10
12 12
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


###**[98]** 설탕과자 뽑기


```
**서론**
부모님과 함께 유원지에 놀러간 영일이는
설탕과자(설탕을 녹여 물고기 등의 모양을 만든 것) 뽑기를 보게 되었다.

길이가 다른 몇 개의 막대를 바둑판과 같은 격자판에 놓는데,
막대에 있는 설탕과자 이름 아래에 있는 번호를 뽑으면 설탕과자를 가져가는 게임이었다.  
(잉어, 붕어, 용 등 여러 가지가 적혀있다.)
```
![대체 텍스트](https://www.codeup.kr/upload/pimg6214_1.png)

격자판의 세로(h), 가로(w), 막대의 개수(n), 각 막대의 길이(l),  
막대를 놓는 방향(d:가로는 0, 세로는 1)과  
막대를 놓는 막대의 가장 왼쪽 또는 위쪽의 위치(x, y)가 주어질 때,  
격자판을 채운 막대의 모양을 출력하는 프로그램을 만들어보자.

**입력**   
첫 줄에 격자판의 세로(h), 가로(w) 가 공백을 두고 입력되고,  
두 번째 줄에 놓을 수 있는 막대의 개수(n)  
세 번째 줄부터 각 막대의 길이(l), 방향(d), 좌표(x, y)가 입력된다.

[입력값의 정의역]  
1 <= w, h <= 100  
1 <= n <= 10  
d = 0 or 1  
1 <= x <= 100-h  
1 <= y <= 100-w
```
5 5
3
2 0 1 1
3 1 2 3
4 1 2 5
```
**출력**  
모든 막대를 놓은 격자판의 상태를 출력한다.  
막대에 의해 가려진 경우 1, 아닌 경우 0으로 출력한다.  
단, 각 숫자는 공백으로 구분하여 출력한다.  
```
1 1 0 0 0
0 0 1 0 1
0 0 1 0 1
0 0 1 0 1
0 0 0 0 1
```

**Tip::**

1. 파이썬에서는 2차원 배열(행렬)을 쉽게 다룰 수 있도록하는 numpy 라이브러리가 있다.
 - [Numpy 배우러가기](https://blog.naver.com/dnfla420/221752730611)
2. 문제에서 요구하는 방식대로 출력하기 위해 '*(Asterisk)'를 사용하였다.
 - ['*(Asterisk)' 배우러가기](https://blog.naver.com/dnfla420/221749146572)


####정답

In [ ]:
h, w = map(int, input().split())
shape = [[0 for _ in range(w)] for _ in range(h)]
n = int(input())

for _ in range(n) :
  l, d, x, y = map(int, input().split())
  x, y = x-1, y-1

  if d == 0 :
    for i in range(l) :
      shape[x][y+i] = 1
  else :
    for i in range(l) :
      shape[x+i][y] = 1

for s in shape :
  print( *s )

5 5
3
2 0 1 1
3 1 2 3
4 1 2 5
1 1 0 0 0
0 0 1 0 1
0 0 1 0 1
0 0 1 0 1
0 0 0 0 1


###**[99]** 성실한 개미


```
**서론**
영일이는 생명과학에 관심이 생겨 왕개미를 연구하고 있었다.

왕개미를 유심히 살펴보던 중 특별히 성실해 보이는 개미가 있었는데,
그 개미는 개미굴에서 나와 먹이까지 가장 빠른 길로 이동하는 것이었다.

개미는 오른쪽으로 움직이다가 벽을 만나면 아래쪽으로 움직여 가장 빠른 길로 움직였다.
(오른쪽에 길이 나타나면 다시 오른쪽으로 움직인다.)

이에 호기심이 생긴 영일이는 그 개미를 미로 상자에 넣고 살펴보기 시작하였다.

미로 상자에 넣은 개미는 먹이를 찾았거나, 더 이상 움직일 수 없을 때까지
오른쪽 또는 아래쪽으로만 움직였다.
```

미로 상자의 구조가 0(갈 수 있는 곳), 1(벽 또는 장애물)로 주어지고,  
먹이가 2로 주어질 때, 성실한 개미의 이동 경로를 예상해보자.

단, 맨 아래의 가장 오른쪽에 도착한 경우, 더 이상 움직일 수 없는 경우, 먹이를 찾은 경우에는
더이상 이동하지 않고 그 곳에 머무른다고 가정한다.


미로 상자의 테두리는 모두 벽으로 되어 있으며,
개미집은 반드시 (2, 2)에 존재하기 때문에 개미는 (2, 2)에서 출발한다.

**입력**   
10*10 크기의 미로 상자의 구조와 먹이의 위치가 입력된다.

```
1 1 1 1 1 1 1 1 1 1
1 0 0 1 0 0 0 0 0 1
1 0 0 1 1 1 0 0 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 1 0 1 0 1
1 0 0 0 0 1 2 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1
```
**출력**  
성실한 개미가 이동한 경로를 9로 표시해 출력한다.
 
```
1 1 1 1 1 1 1 1 1 1
1 9 9 1 0 0 0 0 0 1
1 0 9 1 1 1 0 0 0 1
1 0 9 9 9 9 9 1 0 1
1 0 0 0 0 0 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1
```

**Tip::**  
![대체 텍스트](https://www.codeup.kr/upload/pimg6215_1.png)

1. 문제에서 요구하는 방식대로 출력하기 위해 '*(Asterisk)'를 사용하였다.
 - ['*(Asterisk)' 배우러가기](https://blog.naver.com/dnfla420/221749146572)

**입력 예시 값**
```
ant_house = [
         [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
         [1, 0, 0, 1, 0, 0, 0, 0, 0, 1],
         [1, 0, 0, 1, 1, 1, 0, 0, 0, 1],
         [1, 0, 0, 0, 0, 0, 0, 1, 0, 1],
         [1, 0, 0, 0, 0, 0, 0, 1, 0, 1],
         [1, 0, 0, 0, 0, 1, 0, 1, 0, 1],
         [1, 0, 0, 0, 0, 1, 2, 1, 0, 1],
         [1, 0, 0, 0, 0, 1, 0, 0, 0, 1],
         [1, 0, 0, 0, 0, 0, 0, 0, 0, 1],
         [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
]
```




####정답

In [ ]:
ant_house = []
for i in range(5):
  matrix = list(map(int, input().split()))
  ant_house.append(matrix)

print('===================')

x, y = 1, 1
while ant_house[x][y] != 2:
  if ant_house[x][y] == 0:
    ant_house[x][y] = 9
    y += 1
  else:
    x += 1
    y -= 1
ant_house[x][y] = 9

for house in ant_house :
  print( *house )


1 1 1 1 1 1 1 1 1 1
1 0 0 1 0 0 0 0 0 1
1 0 0 1 1 1 0 0 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 0 0 1 0 1
1 0 0 0 0 1 0 1 0 1
1 0 0 0 0 1 2 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1
1 1 1 1 1 1 1 1 1 1
1 9 9 1 0 0 0 0 0 1
1 0 9 1 1 1 0 0 0 1
1 0 9 9 9 9 9 1 0 1
1 0 0 0 0 0 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 9 1 0 1
1 0 0 0 0 1 0 0 0 1
1 0 0 0 0 0 0 0 0 1
1 1 1 1 1 1 1 1 1 1
